x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -54,10 +54,17 @@ export class MachineLearningFlyout extends Component<FlyoutProps, FlyoutState> {
     hasMLJob: false,
     selectedTransactionType: this.props.urlParams.transactionType
   };
+  public willUnmount = false;
+
+  public componentWillUnmount() {
+    this.willUnmount = true;
+  }
 
   public async componentDidMount() {
     const indexPattern = await getAPMIndexPattern();
-    this.setState({ hasIndexPattern: !!indexPattern });
+    if (!this.willUnmount) {
+      this.setState({ hasIndexPattern: !!indexPattern });
+    }
```

Comment:
There was an issue, where the component could unmount while the index pattern was loading, which caused `setState` to be called on an unmounted component. 
The more correct approach would be to cancel the request but this will do for now.